### DISCUSSION TITLE
Getting mafic working in a cog

### DISCUSSION THREAD

**Discussion by xFGhoul** [ooliver1/mafic#57] [Q&A] [answered]
Been trying to get mafic working for the past couple hours, successfully got it working but an issue I have is with the player. I do not have a way of having a player object i can work with throughout the cog, in terms of `np`, or `pause`, I can't even disconnect the player because there is no player object instantiated, tried using the `MyPlayer` class you gave but it's very wonky. idk. any updated examples?

**Comment by Kraots** [comment, marked as answer]
```py
    @staticmethod
    async def get_player(ctx: Context) -> mafic.Player: 
         if not ctx.guild.voice_client: 
             player = await ctx.author.voice.channel.connect(cls=mafic.Player) 
         else: 
             player = ctx.guild.voice_client 
  
         return player
```

You can use this, basically what it does is it first checks if there's no player for the guild the command was invoked in, and if there isn't it creates one, otherwise it returns the already existing player object for that guild

**Reply by xFGhoul** [reply]
Alright, I'll test this out. Thanks!
